stdlib/public/core/Bit.swift

```diff
@@ -94,11 +94,13 @@ public func < (lhs: Bit, rhs: Bit) -> Bool {
 }
 
 extension Bit : IntegerArithmeticType {
-  static func _withOverflow(v: (Int, overflow: Bool)) -> (Bit, overflow: Bool) {
-    if let b = Bit(rawValue: v.0) {
-      return (b, v.overflow)
+  static func _withOverflow(intResult: Int, overflow: Bool) -> (Bit, overflow: Bool) {
+    if let bit = Bit(rawValue: intResult) {
+      return (bit, overflow: overflow)
     } else {
-      return (Bit(rawValue: v.0 % 2)!, true)
+      let bitRaw = intResult % 2 + (intResult < 0 ? 2 : 0)
```

Comment:
Well, that would rely on the precondition that `intResult >= -2`. This is true for all current calls to the function, but the way it's written now trivially eliminates this precondition.